src/api/dto/get-api.dto.ts

```diff
@@ -0,0 +1,93 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsArray, IsDate, IsEnum, IsNumber, IsString } from 'class-validator';
+import { Status } from 'src/common/enums/apiVerification.enum';
+import { Visibility } from 'src/common/enums/visibility.enum';
+export class GetApiDto {
+  @ApiProperty()
+  @IsString()
+  id: string;
+
+  @ApiProperty()
```

Comment:
You don't need dto to query a db

Reply:
if i add :Promise<Api> i get error for my return statement ( return partialApi;), should i remove the promise type all together?